### PR TITLE
General code fix 2 

### PR DESCRIPTION
**Pull request by SosoTughushi**
his pull request is focused on resolving occurrences of Sonar rules:
squid:UselessParenthesesCheck - Useless parentheses around expressions should be removed to prevent any misunderstanding.
squid:S2293 - The diamond operator ("<>") should be used.

You can find more information about the issue here: 
https://dev.eclipse.org/sonar/rules/show/squid:UselessParenthesesCheck
https://dev.eclipse.org/sonar/rules/show/squid:S2293

Please let me know if you have any questions.

Soso Tughushi

### CODE REVIEW

File: jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java [hunk 1]
```diff
@@ -1254,7 +1254,7 @@ public String toString(int indentFactor) throws JSONException {
 	 * @throws JSONException If the value is or contains an invalid number.
 	 */
 	public static String valueToString(Object value) throws JSONException {
-		if ((value == null) || value.equals(null)) {
```
Comment (maxirosson) [inline jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java]: I would prefer to revert this change. I am not 100% sure that you keep the same behaviour.

File: jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java [hunk 2]
```diff
@@ -1346,7 +1346,7 @@ public Writer write(Writer writer) throws JSONException {
 	
 	static Writer writeValue(Writer writer, Object value, int indentFactor, int indent) throws JSONException,
 			IOException {
-		if ((value == null) || value.equals(null)) {
```
Comment (maxirosson) [inline jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java]: I would prefer to revert this change. I am not 100% sure that you keep the same behaviour.

File: jdroid-android-google-inappbilling/src/main/java/com/android/vending/billing/IInAppBillingService.java [hunk 3]
```diff
@@ -35,12 +35,12 @@ public Stub() {
 		}
```
Comment (maxirosson) [inline jdroid-android-google-inappbilling/src/main/java/com/android/vending/billing/IInAppBillingService.java]: This file is autogenerated, so the changes should be reverted. Thanks

Comment (SosoTughushi) [inline jdroid-android-google-inappbilling/src/main/java/com/android/vending/billing/IInAppBillingService.java]: reverted

File: jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java (hunk 2 above)
Comment (SosoTughushi) [inline jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java]: reverted

File: jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java (hunk 1 above)
Comment (SosoTughushi) [inline jdroid-java/src/main/java/com/jdroid/java/json/JSONObject.java]: reverted
